tests/avifgridapitest.c

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/avif.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+//------------------------------------------------------------------------------
+
+// Creates an image where the pixel values are defined but do not matter.
+// Returns false in case of memory failure.
+static avifBool createImage(int width, int height, int depth, avifPixelFormat yuvFormat, avifImage ** image)
+{
+    *image = avifImageCreate(width, height, depth, yuvFormat);
+    if (*image == NULL) {
+        printf("ERROR: avifImageCreate() failed\n");
+        return AVIF_FALSE;
+    }
+    avifImageAllocatePlanes(*image, AVIF_PLANES_YUV);
+    if (width * height == 0) {
+        return AVIF_TRUE;
+    }
+
+    avifPixelFormatInfo formatInfo;
+    avifGetPixelFormatInfo((*image)->yuvFormat, &formatInfo);
+    uint32_t uvWidth = ((*image)->width + formatInfo.chromaShiftX) >> formatInfo.chromaShiftX;
+    uint32_t uvHeight = ((*image)->height + formatInfo.chromaShiftY) >> formatInfo.chromaShiftY;
+    if (uvWidth < 1) {
+        uvWidth = 1;
+    }
+    if (uvHeight < 1) {
+        uvHeight = 1;
+    }
```

Comment:
I think we should remove these two checks (lines 31-36).

Note that `uvWidth` can only be 0 if `width` is 0.